test/gtest/ucp/test_ucp_request.cc

```diff
@@ -17,97 +17,89 @@ class test_ucp_request : public ucp_test {
     {
         ucp_test::init();
         sender().connect(&receiver(), get_ep_params());
-        int mem_type_pair_index = get_variant_value() % mem_type_pairs.size();
-        m_send_mem_type  = mem_type_pairs[mem_type_pair_index][0];
-        m_recv_mem_type  = mem_type_pairs[mem_type_pair_index][1];
+        int mem_type_pair_index = get_variant_value() %
+                                  supported_mem_types.size();
+        mem_type = supported_mem_types[mem_type_pair_index];
     }
 
     static void get_test_variants(std::vector<ucp_test_variant> &variants)
     {
         int count = 0;
         add_variant(variants, UCP_FEATURE_TAG);
-        for (std::vector<std::vector<ucs_memory_type_t> >::const_iterator iter =
-                 mem_type_pairs.begin(); iter != mem_type_pairs.end(); ++iter) {
-                std::string name =
-                        std::string(ucs_memory_type_names[(*iter)[0]]) + ":" +
-                        std::string(ucs_memory_type_names[(*iter)[1]]);
-                add_variant_with_value(variants, UCP_FEATURE_TAG, count, name);
-                ++count;
-            }
+        std::vector<ucs_memory_type_t>::const_iterator iter;
+
+        for (iter = supported_mem_types.begin();
+             iter != supported_mem_types.end(); ++iter) {
+            std::string name = std::string(ucs_memory_type_names[(*iter)]);
```

Comment:
std::string() is redundant, have implicit ctor
`std::string name = ucs_memory_type_names[*iter]`